app/serializers/subject_selector_serializer.rb

```diff
@@ -0,0 +1,67 @@
+require 'panoptes/restpack_serializer'
+
+class SubjectSelectorSerializer
+  include Panoptes::RestpackSerializer
+  include NoCountSerializer
+
+  attributes :id, :metadata, :locations, :zooniverse_id,
+    :created_at, :updated_at, :href
+
+  optional :retired, :already_seen, :finished_workflow
+
+  preload :locations
+
+  def self.model_class
+    Subject
+  end
+
+  def locations
+    @model.ordered_locations.map do |loc|
+      {
+       loc.content_type => loc.url_for_format(@context[:url_format] || :get)
+      }
+    end
+  end
+
+  def retired
+    @model.retired_for_workflow?(workflow)
+  end
+
+  def already_seen
+    !!(user_seen&.subjects_seen?(@model.id))
+  end
+
+  private
+
+  def include_retired?
+    select_context?
+  end
+
+  def include_already_seen?
+    select_context?
+  end
+
+  def include_finished_workflow?
+    select_context?
+  end
+
+  def select_context?
```

Comment:
Why not just reference that flipper directly in the SubjectSelectorSerializer? I'd much prefer that. And actually, I think the rest of the context should be coming from the controller. Keeps the Subjects::Selector dedicated to selecting subjects, not configuring a serializer.